app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/manualcodeentry/ManualCodeEntryViewModel.kt

```diff
@@ -87,30 +110,51 @@ class ManualCodeEntryViewModel @Inject constructor(
             return
         }
 
-        viewModelScope.launch {
-            authenticatorRepository.createItem(
-                AuthenticatorItemEntity(
-                    id = UUID.randomUUID().toString(),
-                    key = sanitizedCode,
-                    issuer = state.issuer,
-                    accountName = "",
-                    userId = null,
-                    type = if (isSteamCode) {
-                        AuthenticatorItemType.STEAM
-                    } else {
-                        AuthenticatorItemType.TOTP
-                    },
-                    favorite = false,
-                ),
-            )
-            sendEvent(
-                event = ManualCodeEntryEvent.ShowToast(
-                    message = R.string.verification_code_added.asText(),
-                ),
-            )
-            sendEvent(
-                event = ManualCodeEntryEvent.NavigateBack,
-            )
+        if (saveToBitwarden) {
+            // Save to Bitwarden by kicking off save to Bitwarden flow:
+            val didLaunchSaveToBitwarden = authenticatorBridgeManager
+                .startAddTotpLoginItemFlow(
+                    totpUri = "otpauth://totp/?secret=$sanitizedCode&issuer=${state.issuer}",
+                )
+            if (!didLaunchSaveToBitwarden) {
+                mutableStateFlow.update {
+                    it.copy(
+                        dialog = ManualCodeEntryState.DialogState.Error(
+                            title = R.string.something_went_wrong.asText(),
+                            message = R.string.please_try_again.asText(),
+                        ),
+                    )
+                }
+            } else {
+                sendEvent(ManualCodeEntryEvent.NavigateBack)
+            }
```

Comment:
⛏️ I wouldn't be mad if the `if` and `else` block contents were extracted to methods. `detekt` wouldn't either. 😜

Reply:
Nice :) 

I gotta get a lil' better at not instantly suppressing everything. This was a great example of `detekt` being helpful. and of YOU being helpful--thank you!

ed13f4cfeb4a8683649103ff9791decbf0c68e32